mutable_tree.go

```diff
@@ -516,7 +516,7 @@ func (tree *MutableTree) SaveVersion() ([]byte, int64, error) {
 }
 
 func (tree *MutableTree) deleteVersion(version int64) error {
-	if version == 0 {
+	if version <= 0 {
```

Comment:
Is there a tradeoff between this and migrating to uin64? Out of scope for the PR but curious.

Reply:
`uint64` is a breaking change, since we use `int64` in the public APIs.